3DCalendar/ViewController.swift

```diff
@@ -91,42 +91,55 @@ class ViewController: UIViewController, ARSCNViewDelegate {
         path.addLine(to: CGPoint(x: 0.3, y: 0.4))//E
         path.close()
         
-        let shape = SCNShape(path: path, extrusionDepth: 0.2) //20 cm
+        let shape = SCNShape(path: path, extrusionDepth: 0.05) //20 cm
         let color = UIColor.red
         shape.firstMaterial?.diffuse.contents = color
-        shape.chamferRadius = 0.1
+        shape.chamferRadius = 0.0
         //create a node
         let pentagonNode = SCNNode(geometry: shape)
         //position the node and add to the scene
         pentagonNode.position.z = -1
         return pentagonNode
     }
     
+    func createPlane() -> SCNNode {
+        var plane = SCNPlane(width: 0.3, height: 0.3)
+        plane.firstMaterial?.diffuse.contents = UIColor.red
+        let planeNode = SCNNode(geometry: plane)
+        planeNode.position.z = -0.5
+        return planeNode
+    }
+    
     //Test UI paste on top of node
-    func createUIViewOnNode(){
+    func createUIViewOnNode(vector: SCNVector3){
         
         //1. Create An Empty Node
-        let holderNode = createPentagon()
+        let holderNode = shapeNode()
         
         
         //2. Create A New Material
         let material = SCNMaterial()
+        let viewMaterial = SCNMaterial()
         
         //3. Create A UIView As A Holder For Content
-        let viewToAdd = CalendarView(frame: CGRect(x: 100, y: 100, width: 500, height: 500))
-        
+       
         //5. Set The Materials Contents
-        material.diffuse.contents = viewToAdd
+        viewMaterial.diffuse.contents = viewToAdd
+        material.diffuse.contents = UIColor.red
         
         //6. Set The 1st Material Of The Plane
         holderNode.geometry?.firstMaterial = material
-        material.isDoubleSided = true
+        holderNode.childNodes[0].geometry?.firstMaterial = material
+        holderNode.childNodes[1].geometry?.firstMaterial = viewMaterial
+        holderNode.childNodes[2].geometry?.firstMaterial = material
+        holderNode.childNodes[3].geometry?.firstMaterial = material
```

Comment:
We generally want to avoid subscripting because it could lead to a crash if the array isn't as big as we think. No big deal tho! All you have to do is `guard` and check the size of the Array. Or you could `forEach` it